datatree/datatree.py

```diff
@@ -853,6 +853,22 @@ def __setitem__(
         else:
             raise ValueError("Invalid format for key")
 
+    def move(self, original_path: str, to: str) -> DataTree:
+        new_tree = self.copy()
+        target_node = new_tree.__getitem__(original_path)
+        path_to_parent = "/".join(original_path.split("/")[:-1])
+
+        new_tree.__setitem__(to, DataTree(target_node.to_dataset()))
+
+        # if empty str then it's likely the root node
+        if path_to_parent == "":
+            new_tree = new_tree.drop_nodes(str(target_node.name))
```

Comment:
You might want to add a comment explaining why you are dropping nodes here. (I'm not sure why you are either)